openslides_backend/action/actions/projector/add_to_preview.py

```diff
@@ -0,0 +1,55 @@
+from ....models.models import Projection, Projector
+from ....shared.filters import FilterOperator
+from ....shared.patterns import Collection, string_to_fqid
+from ....shared.schema import required_id_schema
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+from ..projection.create import ProjectionCreate
+
+
+@register_action("projector.add_to_preview")
+class ProjectorProject(UpdateAction):
+    """
+    Action to projector project.
+    """
+
+    model = Projector()
+    schema = DefaultSchema(Projection()).get_default_schema(
+        required_properties=["content_object_id"],
+        optional_properties=["options", "stable", "type"],
+        additional_required_fields={
+            "ids": {"type": "array", "items": required_id_schema}
+        },
+        title="Projector project schema",
+    )
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        for instance in action_data:
+            # check content object and get meeting id.
+            content_object = self.datastore.get(
+                string_to_fqid(instance["content_object_id"]), ["meeting_id"]
+            )
+            meeting_id = content_object["meeting_id"]
+
+            # add the preview projections
+            for projector_id in instance["ids"]:
+                max_weight = self.get_max_projection_weight(projector_id)
+                data = {
+                    "meeting_id": meeting_id,
+                    "preview_projector_id": projector_id,
+                    "weight": max_weight + 1,
+                }
+                for field in ("content_object_id", "options", "stable", "type"):
```

Comment:
`content_object_id` is required, you can add it directly to `data`.

Reply:
Done.